stdlib/public/core/Misc.swift

```diff
@@ -110,3 +112,99 @@ public func _getTypeByMangledNameInContext(
   genericContext: UnsafeRawPointer?,
   genericArguments: UnsafeRawPointer?)
   -> Any.Type?
+
+@_silgen_name("swift_demangle")
+internal func _demangleImpl(
+  mangledName: UnsafePointer<Int8>?,
+  mangledNameLength: UInt,
+  outputBuffer: UnsafeMutablePointer<Int8>?,
+  outputBufferSize: UnsafeMutablePointer<UInt>?,
+  flags: UInt32
+) -> UnsafeMutablePointer<Int8>?
+
+/// Given a mangled Swift symbol, demangle it into a human readable format.
+///
+/// Valid Swift symbols begin with the following prefixes:
+///   ┌─────────────────────╥────────┬───────┐
+///   │ Swift Version       ║        │       │
+///   ╞═════════════════════╬════════╪═══════╡
+///   │ Swift 3 and below   ║   _T   │       │
+///   ├─────────────────────╫────────┼───────┤
+///   │ Swift 4             ║  _T0   │       │
+///   ├─────────────────────╫────────┼───────┤
+///   │ Swift 4.x           ║   $S   │  _$S  │
+///   ├─────────────────────╫────────┼───────┤
+///   │ Swift 5+            ║   $s   │  _$s  │
+///   └─────────────────────╨────────┴───────┘
+///
+/// - Parameters:
+///   - mangledName: A mangled Swift symbol.
+/// - Returns: A human readable demangled Swift symbol.
+public func demangle(_ mangledName: String) -> String {
+  return mangledName.utf8CString.withUnsafeBufferPointer {
+    let demangledPtr = _demangleImpl(
+      mangledName: $0.baseAddress,
+      mangledNameLength: UInt($0.count - 1),
+      outputBuffer: nil,
+      outputBufferSize: nil,
+      flags: 0
+    )
+
+    guard demangledPtr != nil else {
+      return mangledName
+    }
+
+    let demangledName = String(cString: demangledPtr!)
+    _swift_stdlib_free(demangledPtr!)
+    return demangledName
+  }
+}
+
+/// Given a mangled Swift symbol, demangle it into a human readable format.
+///
+/// Valid Swift symbols begin with the following prefixes:
+///   ┌─────────────────────╥────────┬───────┐
+///   │ Swift Version       ║        │       │
+///   ╞═════════════════════╬════════╪═══════╡
+///   │ Swift 3 and below   ║   _T   │       │
+///   ├─────────────────────╫────────┼───────┤
+///   │ Swift 4             ║  _T0   │       │
+///   ├─────────────────────╫────────┼───────┤
+///   │ Swift 4.x           ║   $S   │  _$S  │
+///   ├─────────────────────╫────────┼───────┤
+///   │ Swift 5+            ║   $s   │  _$s  │
+///   └─────────────────────╨────────┴───────┘
```

Comment:
The two column view is misleading.  The `_$s` and `$s` are the same.  Darwin has a user label prefix of `_` (as does Windows x86).  The user label prefix should be stripped as appropriate.